### ISSUE TITLE
Combobox dropdown menu is partially blocked by the boundary of the frame

### ISSUE THREAD

**Issue by Krysme** [emilk/egui#1842]
**Describe the bug**

it seems that the drop down menu cannot be outside of the frame, leading to the partially visible items
**To Reproduce**
```rust
        egui::CentralPanel::default().show(ctx, |ui| {
            let combo = ComboBox::from_label("Test").selected_text("Foo");
            combo.show_ui(ui, |ui| {
                ui.selectable_value(&mut self.val, Select::Option1, "Option1");
                ui.selectable_value(&mut self.val, Select::Option2, "Option2");
                ui.selectable_value(&mut self.val, Select::Option3, "Option3");
            });
        });


```
**Expected behavior**
the whole drop down menu and selectable items are totally visible

**Screenshots**

![图片](https://user-images.githubusercontent.com/11013684/180443042-fc13da9a-8590-4ef8-8501-de129e2fc251.png)

**Thoughts**

If it is difficult to implement out of frame controls, is there anyway to have an option so that the menu goes up instead of down? Currently, it seems that `ComboBox` controls cannot be at the bottom of the frame because of this issue.

**Comment by emilk** [comment]
duplicate of https://github.com/emilk/egui/issues/693
